File: c++/Array/Argument.cpp



#include <iostream>

const int size = 5;

void print(int array[size]) {
	for (int i = 0; i < size; ++i) {
		std::cout << array[i] << std::endl;
		++array[i];
	}
}

int main() {
	int array[] = { 0, 1, 2, 3, 4 };

	print(array);

	std::cout << "test" << std::endl;
	for (int i = 0; i < size; ++i) {
		std::cout << array[i] << std::endl;
	}
}

File: Perl/SubRootin/Argument.pl


sub function($@) {
    my @array = $_[1];
    print "test\n";
#    print $temp, "\n";
#    print @array;
    print $array[0], "\n";
#    print "$array[1]";
}


my @array = (1,2);
my %hash = ("test" => 3);
#print $hash{"test"};

&function($hash{"test"}, @array);
print $array[0];
